tests/framework/clients/rancher/catalog/clusterrepo.go

```diff
@@ -57,6 +73,22 @@ func (c *Client) InstallChart(chart *types.ChartInstallAction) error {
 	return result.Error()
 }
 
+// UpgradeChart upgrades the chart according to the parameter `chart`
+func (c *Client) UpgradeChart(chart *types.ChartUpgradeAction) error {
+	bodyContent, err := json.Marshal(chart)
+	if err != nil {
+		return err
+	}
+
+	result := c.RESTClient().Post().
+		AbsPath(rancherChartsURL).Param("action", "upgrade").
+		VersionedParams(&metav1.CreateOptions{}, scheme.ParameterCodec).
+		Body(bodyContent).
+		Do(context.TODO())
+
+	return result.Error()
```

Comment:
Does this function always result in an error, how do we know if it was successful. Are there any intermediate states besides success and error?

will the post wait until the upgrade action is finished or do we need to apply some waiter too?

Reply:
I think this gets answered mostly in the block below

Reply:
If it is successful the error would be nil, and if not an error is returned. The test itself would then do the upgrade validation